### PR TITLE
DO-35 :newspaper: :sparkles: Add placeholder for e2e

### PR DESCRIPTION
**Pull request by dkravetz**
### References

[DO-35](https://xainag.atlassian.net/browse/DO-35)

### Summary

Add placeholder for manual e2e tests on PRs to development branch. Add CI jobs on PRs to master

### Are there any open tasks/blockers for the ticket?

No

---

### Reviewer checklist

*Reviewer agreement:*

* Reviewers assign themselves at the start of the review.
* Reviewers do **not** commit or merge the merge request.
* Reviewers have to check and mark items in the checklist.

**Merge request checklist**

- [ ] Conforms to the merge request title naming `XP-XXX <a description in imperative form>`.
- [ ] Each commit conforms to the naming convention `XP-XXX <a description in imperative form>`.
- [ ] Linked the ticket in the merge request title or the references section.
- [ ] Added an informative merge request summary.

**Code checklist**

- [ ] Conforms to the branch naming `XP-XXX-<a_small_stub>`.
- [ ] Passed scope checks.
- [ ] Added or updated tests if needed.
- [ ] Added or updated code documentation if needed.
- [ ] Conforms to Google docstring style.
- [ ] Conforms to XAIN structlog style.

### CODE REVIEW

File: .github/workflows/development-e2e-tests.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: Comment-based E2E test
+
+on:
+  issue_comment:
+    types: [created, edited]
+
+jobs:
+  e2e:
+    runs-on: ubuntu-18.04
+    if: github.event.comment.body == 'run e2e'
```
Comment (dkravetz) [inline .github/workflows/development-e2e-tests.yml]: Given that GitHub Actions doesn't have manual jobs per se, this is a manual way of triggering them, that doesn't involve sending requests or anything similar.

**Comment by dkravetz** [comment]
run e2e

File: .github/workflows/development-e2e-tests.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+name: Comment-based E2E test
```
Comment (dkravetz) [inline .github/workflows/development-e2e-tests.yml]: run e2e

**Comment by dkravetz** [comment]
run e2e

File: .github/workflows/development-e2e-tests.yml (hunk 2 above)
Comment (dkravetz) [inline .github/workflows/development-e2e-tests.yml]: run e2e

**Comment by dkravetz** [comment]
run e2e

File: .github/workflows/development-e2e-tests.yml (hunk 2 above)
Comment (dkravetz) [inline .github/workflows/development-e2e-tests.yml]: run e2e

**Comment by dkravetz** [comment]
run e2e

**Comment by dkravetz** [comment]
run e2e

**Comment by dkravetz** [comment]
run e2e

**Comment by dkravetz** [comment]
run e2e

**Comment by dkravetz** [comment]
run e2e

**Comment by dkravetz** [comment]
run e2e

**Comment by dkravetz** [comment]
run e2e

**Comment by dkravetz** [comment]
run e2e

**Comment by dkravetz** [comment]
run e2e

File: .github/workflows/development-e2e-tests.yml (hunk 2 above)
Comment (dkravetz) [inline .github/workflows/development-e2e-tests.yml]: run e2e

Comment (dkravetz) [inline .github/workflows/development-e2e-tests.yml]: run e2e

**Comment by dkravetz** [comment]
run e2e

**Comment by dkravetz** [comment]
run e2e

File: .github/workflows/development-e2e-tests.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
```
Comment (dkravetz) [inline .github/workflows/development-e2e-tests.yml]: run e2e

**Comment by dkravetz** [comment]
run e2e

File: .github/workflows/development-e2e-tests.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
```
Comment (dkravetz) [inline .github/workflows/development-e2e-tests.yml]: run e2e

**Comment by dkravetz** [comment]
run e2e

File: .github/workflows/development-e2e-tests.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
```
Comment (dkravetz) [inline .github/workflows/development-e2e-tests.yml]: run e2e

File: .github/workflows/development-e2e-tests.yml (hunk 2 above)
Comment (dkravetz) [inline .github/workflows/development-e2e-tests.yml]: /run e2e

Comment (dkravetz) [reply .github/workflows/development-e2e-tests.yml]: /run e2e

Comment (dkravetz) [reply .github/workflows/development-e2e-tests.yml]: /run e2e

Comment (atymoshchuk) [reply .github/workflows/development-e2e-tests.yml]: /run e2e

File: .github/workflows/master.yml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install --upgrade setuptools
+          pip install -e .[dev]
+
+      - name: black
+        run: |
+          black --check --exclude "xain_fl/cproto/.*_pb2.*" \
+          setup.py conftest.py benchmarks examples xain_fl
```
Comment (atymoshchuk) [inline .github/workflows/master.yml]: there are no benchmarks and examples in the repo any more

File: .github/workflows/master.yml [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+      - name: isort
+        run: |
+          isort --check-only --indent=4 -rc setup.py conftest.py benchmarks \
```
Comment (atymoshchuk) [inline .github/workflows/master.yml]: here as well

File: .github/workflows/master.yml [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+          examples xain_fl
+
+      - name: pylint
+        run: |
+          pylint --rcfile=pylint.ini benchmarks examples xain_fl
```
Comment (atymoshchuk) [inline .github/workflows/master.yml]: and here

File: .github/workflows/master.yml [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+      - name: mypy
+        continue-on-error: true
+        run: |
+          mypy benchmarks examples/* xain_fl
```
Comment (atymoshchuk) [inline .github/workflows/master.yml]: and here

**Pull request merged**